src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -251,6 +253,142 @@ void ConsoleProcess::commonInit()
                            module_context::activeSession().id());
 #endif
    }
+   
+   // if RStudio has been configured to use Python,
+   // and the user has opted in to our terminal config
+   // place that copy of Python on the PATH
+   if (prefs::userPrefs().terminalPythonIntegration())
+   {
+      std::string reticulatePython = core::system::getenv("RETICULATE_PYTHON");
+      if (reticulatePython.empty())
+      {
+         Error error = r::exec::RFunction(".rs.inferReticulatePython")
+               .call(&reticulatePython);
+
+         if (error)
+            LOG_ERROR(error);
+
+         if (!reticulatePython.empty())
+         {
+            core::system::setenv(
+                     &*options_.environment,
+                     "RETICULATE_PYTHON",
+                     reticulatePython);
+         }
+      }
+
+      if (!reticulatePython.empty())
+      {
+         auto isBashShell = [&]()
+         {
+            switch (getShellType())
+            {
+            
+            case TerminalShell::ShellType::PosixBash:
+            case TerminalShell::ShellType::GitBash:
+            case TerminalShell::ShellType::WSLBash:
```

Comment:
I don't see how WSL can work. It's essentially a remote shell into a Linux environment, and the paths are different. You could get into the game of mapping c:\python to /mnt/c/python, I suppose.

The WSL environment may contains its own Linux-native python installation(s).

Perhaps trickiest, the "bash.exe" for launching wsl isn't actually bash (it's a generic launcher; see other comment below about wsl.exe). So it can very well be zsh or something else inside, you don't know (though you might be able to discover such things using various wsl.exe switches; but then you have to worry about which version of Windows-10 it is; some features of wsl aren't in some still-supported Windows-10 versions).

Reply:
If this is the case, then I'd feel more comfortable just leaving WSL out of this PR.